test/functional/api/vim_spec.lua

```diff
@@ -1036,5 +1036,33 @@ describe('api', function()
       eq(expected, nvim("list_uis"))
     end)
   end)
-
+  describe('nvim_get_keyboard_mods_state', function()
+    it('on linux either xlib11 is available or not', function()
+      if os_name() == 'unix' then
+        local status, rv = pcall(nvim, "get_keyboard_mods_state")
+        if status == true then
+          eq(rv, {CapsLock=false, NumLock=false, ScrollLock=false})
```

Comment:
As I said it should not assume specific state of locks. Check for presence of all keys and check for corresponding values being boolean.

Reply:
OK, I thought this was also good, Sure, I replace it